ra_aid/agent_utils.py

```
@@ -544,15 +562,19 @@ def run_agent_with_retry(agent, prompt: str, config: dict) -> Optional[str]:
                             _global_memory['task_completed'] = False
                             _global_memory['completion_message'] = ''
                             break
-                        if _global_memory['task_completed'] or _global_memory['plan_completed']:
+                        if _global_memory['task_completed']:
                             _global_memory['task_completed'] = False
                             _global_memory['completion_message'] = ''
                             break
                     logger.debug("Agent run completed successfully")
                     return "Agent run completed successfully"
```

Ruff (0.8.2)

Consider moving this statement to an `else` block

(TRY300)